### PR TITLE
create delta table if not exists

### PR DESCRIPTION
**Pull request by ZavezX**
## Description

Create delta table for processed timeseries if it doesn't exist.

## References

* #376

### CODE REVIEW

**Review by johevemi — APPROVED** [review]

File: source/databricks/package/timeseries_publisher.py [hunk 1]
```diff
@@ -53,6 +54,9 @@ def publish_timeseries_batch(df, epoch_id, timeseries_processed_path):
             .withColumn("max_month", month("max_time")) \
             .withColumn("max_day", dayofmonth("max_time"))
         row = min_max_df.first()
+
+        create_delta_table_if_empty(timeseries_processed_path, spark)
```
Comment (BjarkeMeier) [inline source/databricks/package/timeseries_publisher.py:58]: I know that this is just 3 more lines but this method is already much less maintainable than desired. Could we rather start refactoring towards a smaller function for better maintainability?

File: source/databricks/package/timeseries_publisher.py [hunk 2]
```diff
@@ -132,3 +136,19 @@ def timeseries_publisher(delta_lake_container_name: str, storage_account_name: s
         option("checkpointLocation", checkpoint_path). \
         foreachBatch(lambda df, epochId: publish_timeseries_batch(df, epochId, timeseries_processed_path)). \
         start()
+
+
+def create_delta_table_if_empty(delta_table_path, spark):
```
Comment (BjarkeMeier) [inline source/databricks/package/timeseries_publisher.py:141]: This looks like something I've seen before. Can we extract a reusable function and reduce duplication?

Comment (ZavezX) [reply source/databricks/package/timeseries_publisher.py:141]: You have seen this in the aggregation domain.

Comment (BjarkeMeier) [reply source/databricks/package/timeseries_publisher.py:141]: Thx. Can we move it to a more reusable and discoverable location (I assume we don't just want to use it in the publisher).

**Pull request merged**